server/src/modules/users/domain/value-objects/user-password.ts

```diff
@@ -1,23 +1,21 @@
-import bcrypt from 'bcrypt'
-
 import { Result } from '../../../../shared/core/result'
-import { ValueObject } from '../../../../shared/domain/value-object'
+import { EncryptedSecret } from './encrypted-secret'
 import { UserValueObjectErrors } from './errors'
 
 export interface UserPasswordProps {
   value: string
   hashed: boolean
 }
 
-export class UserPassword extends ValueObject<UserPasswordProps> {
-  public static minLength: number = 6
+export class UserPassword extends EncryptedSecret {
+  protected static minLength: number = 6
```

Comment:
NIST guidelines say passwords should be at least 8 characters, as mentioned here: https://auth0.com/blog/dont-pass-on-the-new-nist-password-guidelines/

(https://pages.nist.gov/800-63-3/sp800-63b.html section 5.1.1.1 is the exact place where they say so.)

```suggestion
  protected static minLength: number = 8
```